src/ast/passes/codegen_llvm.cpp

```diff
@@ -1900,7 +1901,14 @@ ScopedExpr CodegenLLVM::visit(Call &call)
       arg_values.push_back(args.back().value());
     }
 
-    return ScopedExpr(b_.CreateCall(func, arg_values, call.func));
+    auto *inst = b_.CreateCall(func, arg_values, call.func);
+    return ScopedExpr(inst, [this, inst, &call] {
+      // We set the debug location on the call instructions only after the
+      // scoped expression is not longer used. Otherwise the instruction emitter
+      // seems to use this location for everything, which results in problems.
```

Comment:
Nice trick, I was also fighting this behavior in the past.

Reply:
Yeah, I believe the correct solution is to actually emit the correct debug location on every new node, that we could do with a generic override of the `visit` template for `CodegenLLVM`. E.g. something like:

```
template <typename T>
ScopedExpr visit(T &t)
{
    if (std::is_derived_from_v<Node, T>) {
       b_.SetDebugLocation(createDebugLocation(t.loc(), scope_));
    }
    return codegen(t);
}
```

(We'd also need to change the scope_stack_ to actually generate a new `DILocalScope` for every scope, and pass this in.) But my goal is to actually unlock interop, we can come and clean this up later.